ts/parsers/slideParser.ts

```diff
@@ -107,6 +108,24 @@ export default class SlideParser {
         }
     }
 
+    public static getGroupedNodes(rootGroupNode, groupCount = 0, groupedShapes = [], groupedImages = []) {
+        groupCount++;
+        if (rootGroupNode["p:sp"]) {
+            groupedShapes.push(...rootGroupNode["p:sp"]);
+        }
+        if (rootGroupNode["p:pic"]) {
+            groupedImages.push(...rootGroupNode["p:pic"]);
+        }
+        const subGroups = rootGroupNode["p:grpSp"];
+        if (subGroups && Array.isArray(subGroups) && groupCount <= GROUPS_LIMIT) {
```

Comment:
you can exit early `groupCount <= GROUPS_LIMIT`

Reply:
Clarified with rabi that moving this early will miss out on the last iteration, unless we increase the check to be limit + 1 .. no need for that I think, will leave as is for now.